lib/perl5db.pl

```diff
@@ -2802,9 +2802,13 @@ sub _cmd_l_main {
         return _cmd_l_plus($new_start, $new_incr);
     }
     # l start-stop or l start,stop
-    elsif (my ($s, $e) = $spec =~ /^(?:(-?[\d\$\.]+)(?:[-,]([\d\$\.]+))?)?/ ) {
+    elsif (my ($s, $e) = $spec =~ /\A(?:(\.|\d+)(?:[-,](\.|\d+))?)?\z/ ) {
```

Comment:
Yeah, I passed over the `$i = $spec;` since the tests were passing; your suggestion is, I think, better. I will add the `/a`. At some point the debugger really should be able to debug UTF-8 source code, but this patch is not the pace to do it.

Reply:
I've slightly rearranged the ordering, and cleaned up that confusing nested tertiary, but I'm switching this to your logic and explicitly passing `$current_line`.